### PR TITLE
Improve HCC debug

### PR DESCRIPTION
**Pull request by bensander**
Add DB_KERNARG to print first 128 bytes of kernarg at dispatch.
Track queues by sequnums, for readability.
Print filenname in dispatch test.

### CODE REVIEW

File: lib/hsa/mcwamp_hsa.cpp [hunk 1]
```diff
@@ -39,6 +39,7 @@
 #include <time.h>
 #include <iomanip>
 
+#define KALMAR_DEBUG 0
```
Comment (whchung) [inline lib/hsa/mcwamp_hsa.cpp]: this line seems to be redundant

File: lib/hsa/mcwamp_hsa.cpp [hunk 2]
```diff
@@ -4370,4 +4446,5 @@ extern "C" void PushArgPtrImpl(void *ker, int idx, size_t sz, const void *v) {
       reinterpret_cast<HSADispatch*>(ker);
   void *val = const_cast<void*>(v);
   dispatch->pushPointerArg(val);
+// RUN: %t.out -d 10000 -h %T
```
Comment (whchung) [inline lib/hsa/mcwamp_hsa.cpp]: this line seems to be redundant

File: lib/hsa/mcwamp_hsa.cpp [hunk 3]
```diff
@@ -3445,10 +3483,42 @@ static std::ostream& operator<<(std::ostream& os, const hsa_barrier_and_packet_t
...
+     for (int i=0; i<sizeof(aql)/sizeof(unsigned); i++) {
+         os << " 0x" << std::setw(8) << aqlBytes[i];
+     }
+     os << " ]" << std::dec;
+
    return os;
 }
 
 
+void printKernarg(const void *kernarg_address, int bytesToPrint)
```
Comment (whchung) [inline lib/hsa/mcwamp_hsa.cpp]: maybe adding `static` to avoid potential name collision.

**Review by whchung — CHANGES_REQUESTED** [review]
there are some minor changes required, other than that I'm fine with this PR.

**Comment by bensander** [comment]
Thx Jack, updated Pull request.



On February 17, 2017 at 7:30:10 PM, Wen-Heng (Jack) Chung (notifications@github.com<mailto:notifications@github.com>) wrote:

@whchung requested changes on this pull request.

there are some minor changes required, other than that I'm fine with this PR.

________________________________

In lib/hsa/mcwamp_hsa.cpp<https://github.com/RadeonOpenCompute/hcc/pull/244#discussion_r101878616>:

> @@ -39,6 +39,7 @@
 #include <time.h>
 #include <iomanip>

+#define KALMAR_DEBUG 0


this line seems to be redundant

________________________________

In lib/hsa/mcwamp_hsa.cpp<https://github.com/RadeonOpenCompute/hcc/pull/244#discussion_r101878716>:

> @@ -4370,4 +4446,5 @@ extern "C" void PushArgPtrImpl(void *ker, int idx, size_t sz, const void *v) {
       reinterpret_cast<HSADispatch*>(ker);
   void *val = const_cast<void*>(v);
   dispatch->pushPointerArg(val);
+// RUN: %t.out -d 10000 -h %T


this line seems to be redundant

________________________________

In lib/hsa/mcwamp_hsa.cpp<https://github.com/RadeonOpenCompute/hcc/pull/244#discussion_r101878774>:

>     return os;
 }


+void printKernarg(const void *kernarg_address, int bytesToPrint)


maybe adding static to avoid potential name collision.

**Review by whchung — APPROVED** [review]
LGTM

**Pull request merged**